tests/cases/resource_test.py

```diff
@@ -350,6 +350,83 @@ def testGetResourceById(self):
         self.assertStatusOk(resp)
         self.assertEqual(str(resp.json['_id']), str(self.file1['_id']))
 
+    def testGetResourceByPath(self):
+        self._createFiles()
+
+        # test users
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.admin,
+                            params={'path': '/user/goodlogin'})
+
+        self.assertStatusOk(resp)
+        self.assertEqual(str(resp.json['_id']), str(self.admin['_id']))
+
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.user,
+                            params={'path': '/user/userlogin'})
+        self.assertStatusOk(resp)
+        self.assertEqual(str(resp.json['_id']), str(self.user['_id']))
+
+        # test collections
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.user,
+                            params={'path': '/collection/Test Collection'})
+        self.assertStatusOk(resp)
+        self.assertEqual(str(resp.json['_id']), str(self.collection['_id']))
+
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.admin,
+                            params={'path':
+                                    '/collection/Test Collection/' +
+                                    self.collectionPrivateFolder['name']})
+        self.assertStatusOk(resp)
+        self.assertEqual(str(resp.json['_id']),
+                         str(self.collectionPrivateFolder['_id']))
+
+        # test folders
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.user,
+                            params={'path': '/user/goodlogin/Public'})
+        self.assertStatusOk(resp)
+        self.assertEqual(
+            str(resp.json['_id']), str(self.adminPublicFolder['_id']))
+
+        resp = self.request(path='/resource/lookup',
+                            method='GET', user=self.user,
+                            params={'path': '/user/goodlogin/Private'})
+        self.assertStatus(resp, 403)
```

Comment:
Can we also add a test for the error condition where an invalid path is passed and we get a 400?